tests/common/plugins/custom_skipif/__init__.py

```diff
@@ -0,0 +1,217 @@
+import multiprocessing
+import yaml
+import pytest
+import logging
+import os
+import sys
+import subprocess
+
+
+logger = logging.getLogger()
+
+
+def pytest_collection(session):
+    initialize_cached_variables(session)
+
+
+def initialize_cached_variables(session):
+    custom_skip_if_dict = 'custom_skip_if_dict'
+    dynamic_tests_skip_platform_type = 'dynamic_tests_skip_platform_type'
+    session.config.cache.set(custom_skip_if_dict, None)
+    session.config.cache.set(dynamic_tests_skip_platform_type, None)
+
+
+def pytest_runtest_setup(item):
+    """
+    Skip tests conditionally based on the user_tests_to_be_skipped list
+    """
+    skip_tests_file_path = get_tests_to_be_skipped_path()
+    if os.path.exists(skip_tests_file_path):
+        skip_tests_dict = read_skip_file(item, skip_tests_file_path)
+        platform_type = get_platform_type(item)
+        update_syspath_for_dynamic_import()
+
+        for test_prefix, skip_dict in skip_tests_dict.items():
+            if str(item.nodeid).startswith(test_prefix):
+                logger.debug('Found custom skip condition: {}'.format(test_prefix))
+                skip_checkers_list = prepare_checkers(skip_dict, platform_type)
+                skip_dict_result = run_checkers_in_parallel(skip_checkers_list)
+                make_skip_decision(skip_dict_result, skip_dict)
+
+
+def get_platform_type(item):
```

Comment:
This function should not be in this file, it should be associated with the `Platform` class and called through it. I suspect this is here to avoid calling `show platform summary` on the DUT each time the Platform class is initialized. 

I suggest solving this problem with a cache or some other python mechanism. Having this function in this file is not extensible because other classes may need to make calls to the DUT which do not need to be repeated (i.e. doing `show version` which I plan on implementing) and we want to handle that in a way that does not clutter this file.

Reply:
will move to Platform related file